onadata/apps/api/viewsets/tableau_viewset.py

```diff
@@ -0,0 +1,202 @@
+
+            # Filter out deleted submissions
+            instances = Instance.objects.filter(
+                **qs_kwargs, deleted_at__isnull=True).order_by('pk')
+
+            if count:
+                return Response({'count': instances.count()})
+
+            if should_paginate:
+                instances = self.paginate_queryset(instances)
+
+            data = process_tableau_data(
+                TableauDataSerializer(instances, many=True).data, xform)
+
+            return self.get_streaming_response(data)
+
+        return Response(data)
+
+    @action(methods=['GET'], detail=True)
+    def schema(self, request, **kwargs):
+        self.object = self.get_object()
+        if isinstance(self.object.content_object, XForm):
+            xform = self.object.content_object
+            headers = xform.get_headers(repeat_iterations=1)
+            schemas = defaultdict(dict)
+
+            for child in headers:
+                # Use regex to identify number of repeats
+                repeat_count = len(re.findall(r"\[+\d+\]", child))
+                if re.search(r"\[+\d+\]", child):
+                    table_name = child.split('/')[repeat_count - 1]
+                    table_name = table_name.replace('[1]', '')
+                    repeat_parent_fields = [
+                        '__parent_id',
+                        '__parent_table']
+                    if not schemas[table_name].get('headers'):
+                        schemas[table_name]['headers'] = []
+                        schemas[table_name]['headers'].append(
+                            child.split('/')[repeat_count])
+                        schemas[table_name]['headers'] +=\
+                            repeat_parent_fields
+                    else:
+                        schemas[table_name]['headers'] +=\
+                            [child.split('/')[repeat_count]]
+                    if not schemas[table_name].get('connection_name'):
+                        schemas[table_name]['connection_name'] = (
+                            f"{xform.project_id}_"
+                            f"{xform.id_string}_{table_name}")
+                    elif not schemas[table_name].get('table_alias'):
+                        schemas[table_name]['table_alias'] = table_name
+                else:
+                    if not schemas['data'].get('headers'):
+                        schemas['data']['headers'] = []
+                    if not schemas['data'].get('connection_name'):
+                        schemas['data']['connection_name'] =\
+                            f"{xform.project_id}_{xform.id_string}"
+                    if not schemas['data'].get('table_alias'):
+                        schemas['data']['table_alias'] = "data"
+
+                    # No need to split the repeats down
+                    schemas['data']['headers'].append(child)
+            response_data = [
+                v for k, v in dict(schemas).items()]
```

Comment:
Why do we need to convert `schemas` to a `dict` ?

Reply:
Hey, we do this to convert the defaultdict schema objects we define [here](https://github.com/onaio/onadata/pull/1910/files/ac8c9bd76b6cb0f98047ce2700be6857090320e2#diff-becb9f24a40619aca034de088e182c1ef49420deea38076626b441c765e6809dR161) into a dict object that Zebra expects

Reply:
Would `v for k,v in schemas.items()` not work ?

Reply:
It seems to work too.
I've removed that.